app/controllers/organisation_relationships_controller.rb

```diff
@@ -8,22 +8,72 @@ class OrganisationRelationshipsController < ApplicationController
   def housing_providers
     housing_providers = organisation.housing_providers
     unpaginated_filtered_housing_providers = filtered_collection(housing_providers, search_term)
+    organisations = Organisation.where.not(id: @organisation.id).pluck(:id, :name)
     respond_to do |format|
       format.html do
         @pagy, @housing_providers = pagy(unpaginated_filtered_housing_providers)
+        @organisations = organisations
         @searched = search_term.presence
         @total_count = housing_providers.size
         render "organisation_relationships/housing_providers", layout: "application"
       end
     end
   end
 
+  def add_housing_provider
+    @organisations = Organisation.where.not(id: @organisation.id).pluck(:id, :name)
+    respond_to do |format|
+      format.html do
+        render "organisation_relationships/add_housing_provider", layout: "application"
+      end
+    end
+  end
+
+  def create_housing_provider
+    create(0)
+  end
+
+  def create_managing_agent
+    create(1)
+  end
+
+  def create(relationship_type)
```

Comment:
there's quite a lot of conditional logic, maybe we could just put it in `create_managing_agent` and `create_managing_agent`